libcst/_nodes/_internal.py

```diff
@@ -47,19 +49,39 @@
 @add_slots
 @dataclass(frozen=True)
 class CodePosition:
+    #: Line numbers are 1-indexed.
     line: int
+    #: Column numbers are 0-indexed.
     column: int
 
 
 @add_slots
 @dataclass(frozen=True)
 class CodeRange:
+    # pyre-ignore[13]: Uninitialized attribute
+    #: Starting position of a node.
     start: CodePosition
+    # pyre-ignore[13]: Uninitialized attribute
+    #: Ending position of a node.
     end: CodePosition
 
-    @classmethod
-    def create(cls, start: Tuple[int, int], end: Tuple[int, int]) -> "CodeRange":
-        return CodeRange(CodePosition(start[0], start[1]), CodePosition(end[0], end[1]))
+    @overload
+    def __init__(self, start: CodePosition, end: CodePosition) -> None:
+        ...
+
+    @overload
+    def __init__(self, start: Tuple[int, int], end: Tuple[int, int]) -> None:
```

Comment:
Why don't we just use CodePosition? Then we don't need to overload with tuple.
CodePosition already been used in multiple places.
https://github.com/Instagram/LibCST/blob/a843500b72ff1ea39265ebea840dba1e5ba50ffb/libcst/_nodes/_base.py#L263
https://github.com/Instagram/LibCST/blob/a843500b72ff1ea39265ebea840dba1e5ba50ffb/libcst/_nodes/tests/test_internal.py#L69

Reply:
If we prefer to use tuple to be simple, maybe we can just remove CodePosition?

Reply:
I'm fine with this.

`CodePosition` is convenient for access (it has named `start` and `end` fields), but tuples are convenient for construction.

However, it should still be possible to construct a `CodeRange` with `CodePosition` objects for parity between access and construction, so I think it makes sense to support both in `__init__`.

Reply:
The primary use is for internal testing where it gets very tedious writing out `CodePosition` repeatedly. As Ben said, `CodePosition` is more descriptive when performing an access. Users shouldn't ever need to worry about constructing these objects anyhow as they're generated through the metadata provider.